mistletoe/block_token.py

```diff
@@ -756,6 +756,8 @@ def read(cls, lines):
         while offset < len(string) - 1:
             match_info = cls.match_reference(lines, string, offset)
             if match_info is None:
+                # backtrack the lines that have not been consumed
```

Comment:
Yes, exactly. The backtracking used to be done at multiple points in the match_reference function. But now that the offset (i.e. the position of last consumed content) is so meticulously tracked, it can be done at a single point in the read method instead.

Reply:
OK, thanks for confirmation. :)